axelrod/match.py

```diff
@@ -77,11 +82,28 @@ def play(self):
 
         i.e. One entry per turn containing a pair of actions.
         """
+        if self._prob_end is not None:
```

Comment:
It feels like the statistical details here are not really the concern of the `Match` class -- it just needs to know the value of `end_turn`, not how to generate it. Perhaps it's better to factor this part out (which also has the advantage of allowing other ending schemes), and simply pass `end_turn` to `Match.__init__`.

OTOH perhaps it's better for the calling code to change the value of `turns` (handling the randomization outside the `Match` class). I guess there are several ways one might want to do this...